src/Evolution/Systems/GeneralizedHarmonic/Constraints.cpp

```diff
@@ -467,6 +469,76 @@ void four_index_constraint(
     }
   }
 }
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+Scalar<DataType> constraint_energy(
+    const tnsr::a<DataType, SpatialDim, Frame>& gauge_constraint,
+    const tnsr::a<DataType, SpatialDim, Frame>& f_constraint,
+    const tnsr::ia<DataType, SpatialDim, Frame>& two_index_constraint,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& three_index_constraint,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& four_index_constraint,
```

Comment:
I wanted to bring this up already in the 4-index constraint PR, but somehow didn't.
What do you think of using the anti-symmetry of the first to indices in this variable?

Reply:
Good idea! I've modified the function to take the output of the actual 4-index-constraint function, which is 
a rank-3 tensor containing only the independent components of the 4-index constraint. This lets me also 
eliminate two loops when computing the constraint energy.